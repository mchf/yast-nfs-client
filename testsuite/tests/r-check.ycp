/**
 * Module:
 *   NFS client configuration
 *
 * Summary:
 *   Routines testuite
 *
 * Authors:
 *   Martin Vidner <mvidner@suse.cz>
 *
 * $Id$
 */
{
    // testedfiles: routines.ycp Testsuite.ycp

    include "testsuite.ycp";
    include "nfs/routines.ycp";

    string OK_Name     = "foo.bar.com.tw";
    string TooLongName = "123456789012345678901234567890123456789012345678901234567890";
    string IllegalName = "Something:wrong";

    DUMP ("check_options");
    TEST (``(check_options ("")), [], nil);
    TEST (``(check_options ("defaults")), [], nil);
    TEST (``(check_options ("nolock,bg")), [], nil);
    TEST (``(check_options ("nolock,nobg")), [], nil);
    TEST (``(check_options ("nolock,rsize=8192")), [], nil);
    TEST (``(check_options ("nolock, bg")), [], nil);
    TEST (``(check_options ("nolock,unknownoption")), [], nil);
    TEST (``(check_options ("nolock,unknownassignment=true")), [], nil);
    TEST (``(check_options ("nolock,rsize=")), [], nil);
    TEST (``(check_options ("nolock,two=equal=signs")), [], nil);

    DUMP ("CheckHostName");
    TEST (``(CheckHostName (OK_Name)), [], nil);
    TEST (``(CheckHostName (TooLongName)), [], nil);
    TEST (``(CheckHostName (IllegalName)), [], nil);
    // Too long & illegal char
    TEST (``(CheckHostName (TooLongName + ":")), [], nil);
}
