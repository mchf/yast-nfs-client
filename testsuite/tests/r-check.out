Dump	check_options
Return	Empty options string not allowed!
Return	
Return	
Return	
Return	
Return	Wrong option:  bg
Return	Wrong option: unknownoption
Return	Wrong option: unknownassignment
Return	Empty value for option: rsize
Return	Wrong option: two
Dump	CheckHostName
Return	true
Return	false
Return	false
Return	false
